quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -235,9 +239,18 @@ public Optional<DeprecatedMetadata> getDeprecatedMetadata() {
     private ConfigValue transformValue(String name, String value, ConfigSourceInterceptorContext context, String configSourceName, boolean parentValue) {
         String mappedValue = value;
         
+        boolean mapped = false;
         var theMapper = parentValue ? this.parentMapper : this.mapper;
         if (theMapper != null && (!name.equals(getFrom()) || parentValue)) {
             mappedValue = theMapper.apply(value, context);
+            mapped = true;
+        }
+        
+        // defaults and values from transformers may not have been subject to expansion
+        if ((mapped || configSourceName == null) && mappedValue != null && Expressions.isEnabled() && mappedValue.contains("$")) {
```

Comment:
> Do we control somewhere the Expressions.isEnabled() state?

We don't control it, quarkus does though.

> I just think it might not be necessary to check it here and the condition might be simplified.

No, we are performing a substitute substitution because the value with an expression could have come from a property mapper transformation (for example database urls) and ExpressionConfigSourceInterceptor will no longer apply at this point in the chain. However for safety we don't want to unconditionally substitute - because quarkus turns off substitution when building the default value source.

Reply:
Keeping the `Expressions.isEnabled()` makes it IMHO more robust. We should not make assumptions when it's used.